GH-728: Review 2

utkarshg6 opened the pull request.

utkarshg6 committed:
GH-728: remove unused import in blockchain_interaction_test.rs

utkarshg6 committed:
GH-728: BlockchainBridge receives consuming_wallet instead of an Option<T>

utkarshg6 committed:
GH-728: Check consuming wallet before sending a msg to BlockchainBridge

utkarshg6 committed:
GH-728: remove Option<T> from the wallet_opt param in begin_scan fn

utkarshg6 committed:
GH-728: remove commented out code

utkarshg6 committed:
GH-728: modify AccountantBuilder to make it's field names contain opt

utkarshg6 committed:
GH-728: improve test mocks

utkarshg6 committed:
GH-728: introduce another URL for mumbai testnet

utkarshg6 committed:
GH-728: change the message in trace log

utkarshg6 committed:
GH-728: test drive the NoConsumingWallet Error

utkarshg6 committed:
GH-728: reordering

utkarshg6 committed:
GH-728: add better logging

dnwiebe commented on node/src/accountant/scanners/mod.rs:
```diff
@@ -1320,15 +1317,10 @@ mod tests {
 
         let run_again_result = subject.begin_scan(
-            Some(consuming_wallet),
+            consuming_wallet.clone(),
```
Suspicion: maybe you can remove this `.clone()`.

dnwiebe commented on node/src/accountant/scanners/mod.rs:
```diff
@@ -572,7 +572,7 @@ pub struct PendingPayableScanner {
     fn begin_scan(
         &mut self,
-        _wallet_opt: Option<Wallet>,
+        _consuming_wallet: Wallet,
```
Since this isn't used, it doesn't particularly have to be a consuming wallet. Could you give it a more descriptive name, like `_irrelevant_wallet` or `_dummy_wallet` or something?

dnwiebe commented on node/src/accountant/mod.rs:
```diff
@@ -574,10 +574,22 @@ impl Accountant {
-            self.earning_wallet = wallet_pair.earning_wallet;
-            self.consuming_wallet_opt = Some(wallet_pair.consuming_wallet);
+            if self.earning_wallet != wallet_pair.earning_wallet {
+                debug!(
```
Maybe these should be `info!` logs.

dnwiebe commented on node/src/accountant/mod.rs:
```diff
@@ -849,12 +866,17 @@ impl Accountant {
-        ) {
+        let result = match self.consuming_wallet_opt.clone() {
+            Some(consuming_wallet) => self.scanners.pending_payable.begin_scan(
+                consuming_wallet,
```
Can you either put a comment here explaining that this wallet isn't actually used, or maybe even create a new wallet out of nothing to demonstrate that there's no real dependence on the value of the consuming wallet? (`Wallet::new("not used")`)

dnwiebe commented on node/src/accountant/mod.rs:
```diff
@@ -2128,15 +2158,28 @@ mod tests {
+        let (wallet_2, timestamp_2, response_skeleton_2, logger_2) = second_begin_scan_params;
+        assert_eq!(wallet_1, wallet_2);
+        assert_eq!(wallet_2, earning_wallet);
+        assert!(time_before < timestamp_1);
```
You might want `<=` on these, because it's possible the code could run so quickly that several events get exactly the same timestamp. They should never be in the _wrong_ order, but they might be the same. Different OSs have different timer granularities.

dnwiebe commented on node/src/accountant/mod.rs:
```diff
@@ -2202,14 +2247,30 @@ mod tests {
+        let (wallet_2, timestamp_2, response_skeleton_2, logger_2) = second_begin_scan_params;
+        assert_eq!(wallet_1, wallet_2);
+        assert_eq!(wallet_2, consuming_wallet);
+        assert!(time_before < timestamp_1);
```
`<=`

dnwiebe commented on node/src/accountant/mod.rs:
```diff
@@ -2279,16 +2340,29 @@ mod tests {
+        let (wallet_2, timestamp_2, response_skeleton_2, logger_2) = second_begin_scan_params;
+        assert_eq!(wallet_1, wallet_2);
+        assert_eq!(wallet_2, consuming_wallet);
+        assert!(time_before < timestamp_1);
```
`<=`

dnwiebe commented on node/src/accountant/mod.rs:
```diff
@@ -2279,16 +2340,29 @@ mod tests {
         ));
+        let mut begin_scan_params = begin_scan_params_arc.lock().unwrap();
+        let first_begin_scan_params = begin_scan_params.remove(0);
+        let (wallet_1, timestamp_1, response_skeleton_1, logger_1) = first_begin_scan_params;
```
Bert thinks these wallet names could be better.

Maybe `first_attempt_wallet` and `second_attempt_wallet`. They're the same wallet, just two different calls with that wallet.

utkarshg6 committed:
GH-728: Review 2 changes

utkarshg6 committed:
GH-728: Review 2 leftover changes

utkarshg6 committed:
GH-728: add better assertions for logger

czarte committed:
GH-756 (#426)

* implementing TransactionHashes struct

* fixed test payable_scanner_handles_sent_payable_message

* fixing code for tests

* fixing code for test payable_scanner_panics_when_fingerprints_for_correct_payments_not_found

* last test fixed

* prepared for the code review

* fixed compiler issues

* add rpc server to test masq_erc20_contract_exists_on_polygon_mumbai_integration

* implemented commnets from code review

* fixing last comments from review 2

utkarshg6 committed:
Merge branch 'master' into GH-728

bertllll committed:
GH-781: New Testnet Polygon Amoy (#431)

* GH-781: experimenting with a test utility

* GH-781: task might be complete now

* GH-781: debugging for windows

* GH-781: trying to satisfy Win

* GH-781: clippy

* GH-781: removed bad solution for issue that now a new card was written for

* GH-781: cargo.toml change

* GH-781: review answered

---------

Co-authored-by: Bert <Bert@Bert.com>

kauri-hero committed:
v0.8.0 (#435)

* Update USER-INTERFACE-INTERFACE.md

adjusted RouteFound response format

* Update README.md

updated Readme with badge and adding MASQ Browser

czarte committed:
GH-783 Fix tests failing on non standard environment (#432)

* fixed tests failing on Bert's computer

* formatting

* remove commented out code

* change .clone() to reference

* fixed test for windows in actions: tilde_in_config_file_path_from_commandline_and_args_uploaded_from_config_file

* target_os for dirs crate used only on windows test

* implementing Bert's review comments - change DirsWrapperMock to DirsWrapperReal and simplifying home_dir creation

* fixing windows data-directory assertion in test tilde_in_config_file_path_from_commandline_and_args_uploaded_from_config_file

* fixing home dir assertion for windows

* add condition to use of dirs crate only for windows

* fixing config toml for windows test tilde_in_config_file_path_from_commandline_and_args_uploaded_from_config_file

* fixed windows test

* formatting

* fixed tilde tests for windows

* remove unnecesary function for finding project root, use of current dir for DirsWrapperMock

* adjusting name of test and removing commented out code

utkarshg6 committed:
Merge branch 'master' into GH-728

bertllll committed:
GH-791: Small fixes to stabilize tests in setup_reporter.rs (#443)

* GH-791: fixed

* GH-791: put some explanatory lines back

---------

Co-authored-by: Bert <Bert@Bert.com>

utkarshg6 committed:
Merge branch 'master' into GH-728

utkarshg6 committed:
GH-797: Update min-hops command docs in User Interface (#447)

* GH-797: minor docs modification

* GH-797: some TODOs

* GH-797: add the definitions for min hops and gas price.

* GH-797: remove the markdown code container

utkarshg6 committed:
Merge branch 'master' into GH-728
